src/exo/pyparser.py

```diff
@@ -168,10 +179,12 @@ def result(self):
         return self._cached_result
 
     def push(self):
-        self.locals = self.locals.new_child()
+        if not self.is_fragment:
+            self.locals = self.locals.new_child()
 
     def pop(self):
-        self.locals = self.locals.parents
+        if not self.is_fragment:
+            self.locals = self.locals.parents
```

Comment:
Why doesn't this work when `is_fragment` is true?

Reply:
Because there is no local variables when parsing fragment

Reply:
I think it would be better to initialize `self.locals = srclocals or ChainMap()` in the constructor. We might want to allow matching patterns like `for i in _: x[i] = _` down the line. The push and pop functions are used to control the scopes of for loop variables, too. Having these functions sometimes do nothing will be confusing.

I think it's better to solve this problem with data (init with empty map) than with control (check which AST we're parsing).

Reply:
Hah! You beat me to it!